host.go

```diff
@@ -0,0 +1,188 @@
+package f3
+
+import (
+	"bytes"
+	"context"
+	"time"
+
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/filecoin-project/go-f3/sim"
+	"golang.org/x/xerrors"
+)
+
+type gpbfthost struct {
+	participant *gpbft.Participant
+	manifest    Manifest
+	gpbft.Signer
+	gpbft.Verifier
+	broadcast func(context.Context, []byte) error
+
+	MessageQueue     chan *gpbft.GMessage
+	SelfMessageQueue chan *gpbft.GMessage //for the future when self messages are async
+
+	log Logger
+
+	timer      *time.Timer
+	runningCtx context.Context
+}
+
+func newHost(id gpbft.ActorID, m Manifest, broadcast func(context.Context, []byte) error,
```

Comment:
I'd make `broadcast` an interface so we don't have to change it later. Really, I'd define a single interface like:

```go
type Client interface {
	Signer
	Verifier
	Logger // maybe not?

	BroadcastMessage(context.Context []byte) error
	// receive message? what am I missing?
}
```